doc/volume-permissions-spec.md

```diff
@@ -0,0 +1,67 @@
+### Introduction
+
+Currently the Docker volume driver for VSphere allows any container on any VM with access to a given datastore
+to create, mount, and delete volumes at will. There are no limits to the size of the volumes that
+can be created or quotas for datastore usage for container volumes. This lack of access control
+takes the management of storage out of the hands of the IT administrator and puts it in the hands of
+the developer. This is a worrying proposition to many IT admins, and since they are the ones that
+must allow deployment of the driver and it's associated host agent, we must allow them to regain
+control with proper enforcement mechanisms that limit damage due to malfeasance or mistake.
+
+### Permissions Model
+
+Since docker containers run inside VMs, and can have root permissions, we cannot rely on the VM user
+for access control. Permissions must be granted to a VM, since VMs are created and managed
+by the IT administrator. This brings up a few issues:
+
+  1. How are VMs identified in order to grant permissions?
+  2. What is the granularity of a permission?
+  3. Where are permissions stored?
+  4. What are the actual permissions?
+
+In order to answer these questions, we must consider 2 scenarios. The first scenarios is permissions
+for a single VM with a local datastore, and the second is multiple VMs with a shared datastore. The
+model must work for both scenarios in order to be useful.
+
+1. VMs can share names so must be identified by unique IDs. These IDs should be unique across a
```

Comment:
I was thinking of using the the existing UUIDs. I knew there were 2 of them, but didn't know what the differences were exactly. Thanks @msterin for the tips!  Hopefully the admin won't have to touch these at all. They just need to be maintained in the per-datastore DBs. The only reason I can think of for the admin to use them is if VMs are named the same and the UUIDs are used for disambiguation.